### ISSUE TITLE
How to publish 3dm-gx3-25 linear velocity and acceleration in rostopic

### ISSUE THREAD

**Issue by wlbyyqkn** [LORD-MicroStrain/ROS-MSCL#213]
Hello, I would like to ask the official website program run by ros noetic to publish the linear acceleration and angular acceleration to the terminal.   Is there anything in the program that needs to be changed?

Display after the program is run

... logging to /home/nuc/.ros/log/2565ed0e-746b-11ed-99e5-f336a839863f/roslaunch-nuc-virtual-machine-8760.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.

started roslaunch server http://nuc-virtual-machine:41271/

SUMMARY
========

PARAMETERS
 * /microstrain_inertial_driver/aux_port: /dev/ttyUSB1
 * /microstrain_inertial_driver/baudrate: 115200
 * /microstrain_inertial_driver/debug: False
 * /microstrain_inertial_driver/device_setup: True
 * /microstrain_inertial_driver/diagnostics: False
 * /microstrain_inertial_driver/enable_hardware_odometer: False
 * /microstrain_inertial_driver/filter_adaptive_level: 2
 * /microstrain_inertial_driver/filter_adaptive_time_limit_ms: 15000
 * /microstrain_inertial_driver/filter_aiding_measurement_summary_data_rate: -1
 * /microstrain_inertial_driver/filter_aiding_status_data_rate: -1
 * /microstrain_inertial_driver/filter_angular_zupt: True
 * /microstrain_inertial_driver/filter_angular_zupt_topic: /moving_ang
 * /microstrain_inertial_driver/filter_auto_heading_alignment_selector: 1
 * /microstrain_inertial_driver/filter_auto_init: True
 * /microstrain_inertial_driver/filter_child_frame_id: sensor
 * /microstrain_inertial_driver/filter_data_rate: 10
 * /microstrain_inertial_driver/filter_declination: 0.23
 * /microstrain_inertial_driver/filter_declination_source: 2
 * /microstrain_inertial_driver/filter_dynamics_mode: 1
 * /microstrain_inertial_driver/filter_enable_altimeter_aiding: False
 * /microstrain_inertial_driver/filter_enable_external_gps_time_update: False
 * /microstrain_inertial_driver/filter_enable_external_heading_aiding: False
 * /microstrain_inertial_driver/filter_enable_gnss_antenna_cal: False
 * /microstrain_inertial_driver/filter_enable_gnss_heading_aiding: True
 * /microstrain_inertial_driver/filter_enable_gnss_pos_vel_aiding: True
 * /microstrain_inertial_driver/filter_enable_magnetometer_aiding: False
 * /microstrain_inertial_driver/filter_enable_odometer_aiding: False
 * /microstrain_inertial_driver/filter_enable_vertical_gyro_constraint: False
 * /microstrain_inertial_driver/filter_enable_wheeled_vehicle_constraint: False
 * /microstrain_inertial_driver/filter_external_gps_time_topic: /external_gps_time
 * /microstrain_inertial_driver/filter_external_speed_topic: /external_speed
 * /microstrain_inertial_driver/filter_frame_id: sensor_wgs84
 * /microstrain_inertial_driver/filter_gnss_antenna_cal_max_offset: 0.1
 * /microstrain_inertial_driver/filter_gnss_dual_antenna_data_rate: -1
 * /microstrain_inertial_driver/filter_heading_data_rate: -1
 * /microstrain_inertial_driver/filter_heading_source: 1
 * /microstrain_inertial_driver/filter_heading_state_data_rate: -1
 * /microstrain_inertial_driver/filter_imu_data_rate: -1
 * /microstrain_inertial_driver/filter_init_attitude: [0.0, 0.0, 0.0]
 * /microstrain_inertial_driver/filter_init_condition_src: 0
 * /microstrain_inertial_driver/filter_init_position: [0.0, 0.0, 0.0]
 * /microstrain_inertial_driver/filter_init_reference_frame: 2
 * /microstrain_inertial_driver/filter_init_velocity: [0.0, 0.0, 0.0]
 * /microstrain_inertial_driver/filter_lever_arm_offset: [0.0, 0.0, 0.0]
 * /microstrain_inertial_driver/filter_nmea_gga_data_rate: 0
 * /microstrain_inertial_driver/filter_nmea_gll_data_rate: 0
 * /microstrain_inertial_driver/filter_nmea_hdt_data_rate: 0
 * /microstrain_inertial_driver/filter_nmea_prka_data_rate: 0
 * /microstrain_inertial_driver/filter_nmea_rmc_data_rate: 0
 * /microstrain_inertial_driver/filter_nmea_talker_id: 3
 * /microstrain_inertial_driver/filter_odom_data_rate: -1
 * /microstrain_inertial_driver/filter_pps_source: 1
 * /microstrain_inertial_driver/filter_relative_odom_data_rate: -1
 * /microstrain_inertial_driver/filter_relative_position_config: False
 * /microstrain_inertial_driver/filter_relative_position_frame: 2
 * /microstrain_inertial_driver/filter_relative_position_ref: [0, 0, 0.01]
 * /microstrain_inertial_driver/filter_reset_after_config: True
 * /microstrain_inertial_driver/filter_sensor2vehicle_frame_selector: 1
 * /microstrain_inertial_driver/filter_sensor2vehicle_frame_transformation_euler: [0.0, 0.0, 0.0]
 * /microstrain_inertial_driver/filter_sensor2vehicle_frame_transformation_matrix: [1.0, 0.0, 0.0, 0...
 * /microstrain_inertial_driver/filter_sensor2vehicle_frame_transformation_quaternion: [0.0, 0.0, 0.0, 1.0]
 * /microstrain_inertial_driver/filter_speed_lever_arm: [0.0, 0.0, 0.0]
 * /microstrain_inertial_driver/filter_status_data_rate: -1
 * /microstrain_inertial_driver/filter_use_compensated_accel: True
 * /microstrain_inertial_driver/filter_vel_in_vehicle_frame: True
 * /microstrain_inertial_driver/filter_velocity_zupt: True
 * /microstrain_inertial_driver/filter_velocity_zupt_topic: /moving_vel
 * /microstrain_inertial_driver/gnss1_antenna_offset: [0.0, -0.7, -1.0]
 * /microstrain_inertial_driver/gnss1_data_rate: 2
 * /microstrain_inertial_driver/gnss1_fix_info_data_rate: -1
 * /microstrain_inertial_driver/gnss1_frame_id: gnss1_antenna_wgs84
 * /microstrain_inertial_driver/gnss1_nav_sat_fix_data_rate: -1
 * /microstrain_inertial_driver/gnss1_nmea_gga_data_rate: 0
 * /microstrain_inertial_driver/gnss1_nmea_gll_data_rate: 0
 * /microstrain_inertial_driver/gnss1_nmea_gsv_data_rate: 0
 * /microstrain_inertial_driver/gnss1_nmea_hdt_data_rate: 0
 * /microstrain_inertial_driver/gnss1_nmea_rmc_data_rate: 0
 * /microstrain_inertial_driver/gnss1_nmea_talker_id: 1
 * /microstrain_inertial_driver/gnss1_nmea_vtg_data_rate: 0
 * /microstrain_inertial_driver/gnss1_nmea_zda_data_rate: 0
 * /microstrain_inertial_driver/gnss1_odom_data_rate: -1
 * /microstrain_inertial_driver/gnss1_rf_error_detection_data_rate: -1
 * /microstrain_inertial_driver/gnss1_sbas_info_data_rate: -1
 * /microstrain_inertial_driver/gnss1_time_reference_data_rate: -1
 * /microstrain_inertial_driver/gnss2_antenna_offset: [0.0, 0.7, -1.0]
 * /microstrain_inertial_driver/gnss2_data_rate: 2
 * /microstrain_inertial_driver/gnss2_fix_info_data_rate: -1
 * /microstrain_inertial_driver/gnss2_frame_id: gnss2_antenna_wgs84
 * /microstrain_inertial_driver/gnss2_nav_sat_fix_data_rate: -1
 * /microstrain_inertial_driver/gnss2_nmea_gga_data_rate: 0
 * /microstrain_inertial_driver/gnss2_nmea_gll_data_rate: 0
 * /microstrain_inertial_driver/gnss2_nmea_gsv_data_rate: 0
 * /microstrain_inertial_driver/gnss2_nmea_hdt_data_rate: 0
 * /microstrain_inertial_driver/gnss2_nmea_rmc_data_rate: 0
 * /microstrain_inertial_driver/gnss2_nmea_talker_id: 2
 * /microstrain_inertial_driver/gnss2_nmea_vtg_data_rate: 0
 * /microstrain_inertial_driver/gnss2_nmea_zda_data_rate: 0
 * /microstrain_inertial_driver/gnss2_odom_data_rate: -1
 * /microstrain_inertial_driver/gnss2_rf_error_detection_data_rate: -1
 * /microstrain_inertial_driver/gnss2_sbas_info_data_rate: -1
 * /microstrain_inertial_driver/gnss2_time_reference_data_rate: -1
 * /microstrain_inertial_driver/gpio1_behavior: 0
 * /microstrain_inertial_driver/gpio1_feature: 0
 * /microstrain_inertial_driver/gpio1_pin_mode: 0
 * /microstrain_inertial_driver/gpio2_behavior: 0
 * /microstrain_inertial_driver/gpio2_feature: 0
 * /microstrain_inertial_driver/gpio2_pin_mode: 0
 * /microstrain_inertial_driver/gpio3_behavior: 0
 * /microstrain_inertial_driver/gpio3_feature: 0
 * /microstrain_inertial_driver/gpio3_pin_mode: 0
 * /microstrain_inertial_driver/gpio4_behavior: 0
 * /microstrain_inertial_driver/gpio4_feature: 0
 * /microstrain_inertial_driver/gpio4_pin_mode: 0
 * /microstrain_inertial_driver/gpio_config: False
 * /microstrain_inertial_driver/gps_leap_seconds: 18.0
 * /microstrain_inertial_driver/imu_angular_cov: [0.01, 0, 0, 0, 0...
 * /microstrain_inertial_driver/imu_data_rate: 100
 * /microstrain_inertial_driver/imu_frame_id: sensor
 * /microstrain_inertial_driver/imu_gps_corr_data_rate: -1
 * /microstrain_inertial_driver/imu_linear_cov: [0.01, 0, 0, 0, 0...
 * /microstrain_inertial_driver/imu_mag_data_rate: -1
 * /microstrain_inertial_driver/imu_nmea_prkr_data_rate: 0
 * /microstrain_inertial_driver/imu_orientation_cov: [0.01, 0, 0, 0, 0...
 * /microstrain_inertial_driver/imu_overrange_status_data_rate: -1
 * /microstrain_inertial_driver/imu_raw_data_rate: -1
 * /microstrain_inertial_driver/nmea_frame_id: nmea
 * /microstrain_inertial_driver/nmea_message_allow_duplicate_talker_ids: False
 * /microstrain_inertial_driver/nmea_message_config: False
 * /microstrain_inertial_driver/odometer_scaling: 0.0
 * /microstrain_inertial_driver/odometer_uncertainty: 0.0
 * /microstrain_inertial_driver/poll_max_tries: 60
 * /microstrain_inertial_driver/poll_port: False
 * /microstrain_inertial_driver/poll_rate_hz: 1.0
 * /microstrain_inertial_driver/port: /dev/ttyUSB0
 * /microstrain_inertial_driver/publish_nmea: False
 * /microstrain_inertial_driver/raw_file_directory: /home/your_name
 * /microstrain_inertial_driver/raw_file_enable: False
 * /microstrain_inertial_driver/raw_file_include_support_data: False
 * /microstrain_inertial_driver/rtcm_topic: /rtcm
 * /microstrain_inertial_driver/rtk_data_rate: 1
 * /microstrain_inertial_driver/rtk_dongle_enable: False
 * /microstrain_inertial_driver/rtk_status_data_rate: 1
 * /microstrain_inertial_driver/save_settings: False
 * /microstrain_inertial_driver/sbas_apply_integrity: False
 * /microstrain_inertial_driver/sbas_enable: True
 * /microstrain_inertial_driver/sbas_enable_corrections: True
 * /microstrain_inertial_driver/sbas_enable_ranging: False
 * /microstrain_inertial_driver/set_baud: False
 * /microstrain_inertial_driver/subscribe_rtcm: False
 * /microstrain_inertial_driver/use_device_timestamp: False
 * /microstrain_inertial_driver/use_enu_frame: False
 * /microstrain_inertial_driver/use_ros_time: False
 * /rosdistro: noetic
 * /rosversion: 1.15.15

NODES
  /
    microstrain_inertial_driver (microstrain_inertial_driver/microstrain_inertial_driver_node)

auto-starting new master
process[master]: started with pid [8768]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to 2565ed0e-746b-11ed-99e5-f336a839863f
process[rosout-1]: started with pid [8778]
started core service [/rosout]
process[microstrain_inertial_driver-2]: started with pid [8785]
[ INFO] [1670223906.368450061]: Running microstrain_inertial_driver version: 2.7.1-8-g04a1cdb
[ INFO] [1670223906.369346336]: Using MIP SDK version: v1.0.0-21-gb1e6278
[ INFO] [1670223906.374364983]: Attempting to open serial port </dev/ttyUSB0> at <115200>
[ INFO] [1670223910.679198758]: Setting device to idle in order to configure
[ INFO] [1670223913.699851658]: Main Connection Info:
    #######################
    Model Name:       3DM-GX3-25
    Serial Number:    6223.38168
    Firmware Version: 2.0.06
    #######################
[ INFO] [1670223913.700900230]: Note: Not opening aux port because RTK dongle enable was not set to true.
[ INFO] [1670223913.745877149]: Note: The device does not support publishing the topic gnss1/aiding_status
[ INFO] [1670223913.841872935]: Note: The device does not support publishing the topic gnss1/fix
[ INFO] [1670223913.939027048]: Note: The device does not support publishing the topic gnss1/fix_info
[ INFO] [1670223914.122864757]: Note: The device does not support publishing the topic gnss1/odom
[ INFO] [1670223914.170890867]: Note: The device does not support publishing the topic gnss1/rf_error_detection
[ INFO] [1670223914.214895606]: Note: The device does not support publishing the topic gnss1/sbas_info
[ INFO] [1670223914.302873405]: Note: The device does not support publishing the topic gnss1/time_ref
[ INFO] [1670223914.345947154]: Note: The device does not support publishing the topic gnss2/aiding_status
[ INFO] [1670223914.389908972]: Note: The device does not support publishing the topic gnss2/fix
[ INFO] [1670223914.434973624]: Note: The device does not support publishing the topic gnss2/fix_info
[ INFO] [1670223914.516089541]: Note: The device does not support publishing the topic gnss2/odom
[ INFO] [1670223914.555969404]: Note: The device does not support publishing the topic gnss2/rf_error_detection
[ INFO] [1670223914.595979062]: Note: The device does not support publishing the topic gnss2/sbas_info
[ INFO] [1670223914.635979622]: Note: The device does not support publishing the topic gnss2/time_ref
[ INFO] [1670223914.675990121]: Note: The device does not support publishing the topic gps_corr
[ INFO] [1670223914.796012902]: Note: The device does not support publishing the topic imu/data
[ INFO] [1670223914.836029482]: Note: The device does not support publishing the topic imu/overrange_status
[ INFO] [1670223914.883025366]: Note: The device does not support publishing the topic mag
[ INFO] [1670223914.924022179]: Note: The device does not support publishing the topic nav/aiding_summary
[ INFO] [1670223914.964056252]: Note: The device does not support publishing the topic nav/dual_antenna_status
[ INFO] [1670223915.165090412]: Note: The device does not support publishing the topic nav/filtered_imu/data
[ INFO] [1670223915.205123813]: Note: The device does not support publishing the topic nav/heading
[ INFO] [1670223915.245123261]: Note: The device does not support publishing the topic nav/heading_state
[ INFO] [1670223915.527226990]: Note: The device does not support publishing the topic nav/odom
[ INFO] [1670223915.811244216]: Note: The device does not support publishing the topic nav/relative_pos/odom
[ INFO] [1670223915.856364061]: Note: The device does not support publishing the topic nav/status
[ INFO] [1670223915.902289630]: Note: The device does not support publishing the topic rtk/status
[ INFO] [1670223915.944247602]: Note: The device does not support publishing the topic rtk/status_v1
[ INFO] [1670223915.968428438]: Note: The device does not support the GPIO config command
[ INFO] [1670223915.990290472]: Note: The device does not support the PPS source command
[ INFO] [1670223916.012310654]: Note: The device does not support the odometer settings command
[ INFO] [1670223916.035282365]: Note: The device does not support the factory streaming channels setup command
[ INFO] [1670223916.057294743]: Note: The device does not support the SBAS settings command
[ INFO] [1670223916.077359108]: Note: The device does not support the nmea message format command
[ INFO] [1670223916.098374903]: Note: Device does not support the RTK dongle config command
[ INFO] [1670223916.132357207]: Note: Device does not support the declination source command.
[ INFO] [1670223916.174355091]: Note: Device does not support GNSS antenna offsets
[ INFO] [1670223916.194362680]: Note: The device does not support the vehicle dynamics mode command.
[ INFO] [1670223916.214457694]: Note: The device does not support the heading source command.
[ INFO] [1670223916.236521239]: Note: The device does not support the filter autoinitialization command.
[ INFO] [1670223916.256378773]: Note: The device does not support the filter adaptive settings command.
[ INFO] [1670223916.276399799]: Note: The device does not support the filter aiding command.
[ INFO] [1670223916.298405614]: Note: The device does not support the relative position configuration command
[ INFO] [1670223916.318387364]: Note: The device does not support the filter speed lever arm command.
[ INFO] [1670223916.338399120]: Note: The device does not support the wheeled vehicle constraint command.
[ INFO] [1670223916.358457284]: Note: The device does not support the vertical gyro constraint command.
[ INFO] [1670223916.378536667]: Note: The device does not support the GNSS antenna calibration command.
[ INFO] [1670223916.398431934]: Note: The device does not support the next-gen filter initialization command.
[ WARN] [1670223916.438506508]: Note: The device does not support the sensor to vehicle transformation or rotation euler command
[ INFO] [1670223916.458475569]: Note: The device does not support the reference point lever arm command
[ INFO] [1670223916.458551953]: Note: The settings were not saved as startup settings. Power cycling will remove changes from device
[ WARN] [1670223916.480484530]: Device does not support the filter reset command
[ INFO] [1670223917.680722588]: Setting spin rate to <1.000000> hz
[ INFO] [1670223917.724801186]: Resuming the device data streams
[ INFO] [1670223917.729825691]: Starting Data Parsing

**Comment by robbiefish** [comment]
@wlbyyqkn Thank you for bringing this to our attention. This should be fixed in #216 for ROS and #215 for ROS2. If possible, could you test on these branches and let me know if that fixes your issue? Otherwise, they should be merged by the end of the day

**Comment by wlbyyqkn** [comment]
Of course, thank you very much for your answer.&nbsp;If you can update&nbsp;complete&nbsp;, please let me know or send me the link.&nbsp;Thank you very much！&nbsp;
&nbsp;


curtain?
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2022年12月5日(星期一) 晚上11:06
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [LORD-MicroStrain/microstrain_inertial] How to publish 3dm-gx3-25 linear velocity and acceleration in rostopic (Issue #213)





 
@wlbyyqkn Thank you for bringing this to our attention. This should be fixed in #216 for ROS and #215 for ROS2. If possible, could you test on these branches and let me know if that fixes your issue? Otherwise, they should be merged by the end of the day

**Comment by robbiefish** [comment]
@wlbyyqkn the fix has been merged into the main branches. Please update your clone with:
```bash
git pull origin ros --recurse-submodules
```
or if using ROS2
```bash
git pull origin ros2 --recurse-submodules
```
